sklearn/model_selection/tests/test_validation.py

```diff
@@ -733,6 +734,27 @@ def test_permutation_test_score_allow_nans():
     permutation_test_score(p, X, y, cv=5)
 
 
+def test_permutation_test_score_fit_params():
+    est = SVC(random_state=0)
+    n_samples = 50
+    X, y = make_blobs(n_samples=n_samples, n_features=2, centers=10,
+                      random_state=0)
+    # sample_weight is a list
+    w = [i for i in range(n_samples)]
+    l_with_fit_params = permutation_test_score(est, X, y,
+                                               fit_params={'sample_weight': w})
+    l_without_fit_params = permutation_test_score(est, X, y)
+    l_without_fit_params_m1 = l_without_fit_params[1].mean()
+    assert not np.isclose(l_with_fit_params[1].mean(),
+                          l_without_fit_params_m1)
+    # sample_weight is an array
+    W = np.array(w)
```

Comment:
don't use capitals for 1d arrays.

Also, why not just do a loop over [list, np.array] and simply cast the original w?